cpp/tensorrt_llm/kernels/cutlass_kernels/moe_gemm/moe_kernels.cu

```diff
@@ -2292,6 +2292,8 @@ void doActivation(T* output, GemmOutputType const* gemm_result, float const* fp8
 
     auto fn = [&]()
     {
+        // IMPORTANT: Keep the order of the activation functions in the same order as the ActivationType enum in
+        // common.h
```

Comment:
Let convert this to std::map{} and index with `map.at()` to avoid this issue in the future.
e.g. 
```
auto fn_map = std::map{
    {ActivationType::Gelu, &doActivationKernel<T, GemmOutputType, ScaleBiasType,  IdentityAdaptor<cutlass::epilogue::thread::GELU>, decltype(block_scaling_type)::value>}, 
    {ActivationType::Relu, &doActivationKernel<T, GemmOutputType, ScaleBiasType,  IdentityAdaptor<cutlass::epilogue::thread::ReLu>, decltype(block_scaling_type)::value>},
... /// etc for the other types
}
return fn_map.at(activation_type.activation_type);
```